ntbs-service/Services/SpecimenService.cs

```diff
@@ -111,6 +114,29 @@ public async Task<IEnumerable<SpecimenMatchPairing>> GetAllSpecimenPotentialMatc
             }
         }
 
+        public async Task<IEnumerable<(string LegacyId, string ReferenceLaboratoryNumber)>>
+            GetLegacyReferenceLaboratoryMatches(IEnumerable<string> legacyIds)
+        {
+            // The table we're referencing here has legacyIds stored as INTs (since they are all ETS ids)
+            // Therefore we need to convert to and from strings
+            var intIds = legacyIds
+                .Select(id => int.TryParse(id, out var intId) ? intId : (int?)null)
+                .Where(id => id.HasValue)
+                .Select((id => id.Value));

```

Comment:
I know this isn't your code, but there's an unnecessary set of brackets here.

Reply:
Fixed!